apps/api/src/scraper/scrapeURL/engines/pdf/index.ts

```diff
@@ -147,11 +150,21 @@ async function scrapePDFWithRunPodMU(
 async function scrapePDFWithParsePDF(
   meta: Meta,
   tempFilePath: string,
+  maxPages?: number,
 ): Promise<PDFProcessorResult> {
   meta.logger.debug("Processing PDF document with parse-pdf", { tempFilePath });
 
   const result = await PdfParse(await readFile(tempFilePath));
-  const escaped = escapeHtml(result.text);
+  let text = result.text;
+  
+  if (maxPages !== undefined && result.numpages > maxPages) {
+    const lines = text.split('\n');
+    const estimatedLinesPerPage = Math.ceil(lines.length / result.numpages);
+    const maxLines = maxPages * estimatedLinesPerPage;
+    text = lines.slice(0, maxLines).join('\n');
+  }
+  
+  const escaped = escapeHtml(text);
 
```

Comment:
ignore maxPages if using pdfparse, it doesn't cost us anything to do it if this method is used.